### PR TITLE
Disease/help strings

### PR DESCRIPTION
**Pull request by j3b4**
Edited a copy of help_entries/races.py  to create help_entries/diseases.py.  Then I started adding the doc strings to the empty classes in world/diseases.py .  No idea what will go on behind the scene.

I'm using UESRPG text for the disease descriptions but have taken the liberty to make the mechanics more vague so as not to give players inaccurate or distracting information.  I could change this to just put in the numbers from the RPG verbatim ie.  -50 Magicka  ; and roll d100 after one week.   

Let me know what you think.

### CODE REVIEW

**Comment by duysqubix** [comment]
I will check it out in the morning to see if it updates fine. From the looks of it, it should work as expected.

**Comment by duysqubix** [comment]
Immediately what I can see though is that you will have to add a module level variable of 

```
ALL_DISEASES
```

That is a list of the class names. Such as 

```python
ALL_DISEASES = [Ataxia, Brainrot,...]

```

So that the wrapper can iterate through the static classes and access the \_\_doc\_\_ string

**Review by duysqubix — CHANGES_REQUESTED** [review]
Need to add an ALL_DISEASES list here.

File: resources/help_entries/diseases.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+main_entrytxt = \
+f"""
+
+{diseases}
+
+more help is available on the them by |ghelp <disease>|n
+"""
+DiseaseHelpEntry.create_main(HELP_CATEGORY, main_entrytxt)
+
+for sign in ALL_DISEASES:
```
Comment (duysqubix) [inline resources/help_entries/diseases.py:37]: I think you meant to use disease instead of sign

File: resources/help_entries/diseases.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+HELP_CATEGORY = 'disease'
+
+
+class RaceSignHelpEntry(HelpEntryWrapper):
+    __help_category__ = HELP_CATEGORY
+
+
+# CODE
+
+diseases = ", ".join([x.name.capitalize() for x in ALL_DISEASES])
```
Comment (duysqubix) [inline resources/help_entries/diseases.py:26]: In order to get the name of all diseases you need to have a way of actually getting it. 

For signs (which I see you copy/pasted from) the list of all signs is initialized which is why I you can access the property of `name` 

Diseases doesn't have that (not yet anyways) so you could access class name variable like `__name__` such as 

```python
diseases = [x.__name__ for x in ALL_DISEASES]
```

**Review by duysqubix — CHANGES_REQUESTED** [review]

**Comment by j3b4** [comment]
Is something like this okay:   

`class BoneBreak(Disease):
    """
    """
    __obj_name__ = "bone break fever"
    pass`

_obj_name__ is what is used in world/races.py and world/birthsigns.py and in help_entries both races and birthsigns use  string = ", ".join([x.name.capitalize()

Just wondering if I should follow your example in the comment above or in your code.

**Comment by duysqubix** [comment]
Oh yea if I do that then go for it. Help entries is one of the first things I did quite a while ago. Memory is a bit rusty.

**Comment by duysqubix** [comment]
Finally at a computer @j3b4 
I see that I haven't even touched Diseases yet. You can continue using the same pattern like I do for Races and Conditions by using

`__obj_name__` if you plan on using the code of 

```python
diseases = [x.name.capitalize() for x in ALL_DISEASES]
```

Then you will need to have the following done for the parent class of Disease something like this:

```python

class Disease:
    __obj_name__ = ""
   
   @property
   def name(self):
      return self.__obj_name__

class BrainRot(Disease):
   __obj_name__ = "brain_rot" #try to refrain from using spaces, replace with _

class Fever(Disease):
   __obj_name__ = "fever"

ALL_DISEASES = [BrainRot(), Fever()]

```

**Comment by j3b4** [comment]
Okay this completes the docstrings for diseases. The main descriptions are verbatim (except for typos I noticed in the original) UESRPG 3e (version 3) but "symptoms" and "progression" have been edited to avoid describing mechanics that may or may not be reflected in the MUD code.

**Comment by duysqubix** [comment]
Ha. The coronavirus was a nice touch.

**Comment by j3b4** [comment]
Added the name method to Disease() and replaced spaces with underscores in names.

File: resources/help_entries/diseases.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+from evennia import create_help_entry
+from evennia.help.models import HelpEntry
+from evennia.utils import dedent
+from world.diseases import ALL_DISEASES  # TODO: actually create that there
+from resources.help_entries.help_entry_wrapper import HelpEntryWrapper
+
+HELP_CATEGORY = 'disease'
+
+
+class RaceSignHelpEntry(HelpEntryWrapper):
```
Comment (duysqubix) [inline resources/help_entries/diseases.py:20]: Think you meant **DiseaseHelpEntry**

File: world/diseases.py [hunk 4]
```diff
@@ -1,2 +1,565 @@
...
+    pass
+
+class CoronaVirus(Disease)
+    """
+    This easter egg disease has yet to be seen in Tamriel thanks 
+    entirely to the Emperor's strict policy prohibiting visitors
+    from twenty-first century Earth. However if it arrives it 
+    may have a devastating impact. Adventurers are advised to 
+    "work from home" and avoid taverns and churches unless masked.
+
```
Comment (duysqubix) [inline world/diseases.py:543]: Missing a ending block comment here

"""

File: world/diseases.py [hunk 5]
```diff
@@ -1,2 +1,565 @@
...
+    Symptoms
+    --------
+    Mild respiratory impairment or death. Exacerbation of political 
+    turmoil.
+
+    Carriers
+    --------
+    Other people - especially those you disagree with;
+    Cliff Racers
```
Comment (duysqubix) [inline world/diseases.py:552]: Ha. Cliff racers

**Comment by duysqubix** [comment]
Very nice. I'll checkout latest to see if it successfully updates help entries on diseases.

**Comment by duysqubix** [comment]
So I have made some minor changes to the PR to have it work properly.

I would recommend removing the sections in the `diseases` it messes up the formatting.

**Review by duysqubix — APPROVED** [review]

**Pull request merged**